django_saml2_auth/views.py

```diff
@@ -184,12 +186,25 @@ def acs(r):
         if settings.SAML2_AUTH.get('TRIGGER', {}).get('BEFORE_LOGIN', None):
             import_string(settings.SAML2_AUTH['TRIGGER']['BEFORE_LOGIN'])(user_identity)
     except User.DoesNotExist:
+        logger.info(f'not found user with {User.USERNAME_FIELD} {user_name}')
         new_user_should_be_created = settings.SAML2_AUTH.get('CREATE_USER', True)
         if new_user_should_be_created:
-            target_user = _create_new_user(user_name, user_email, user_first_name, user_last_name)
-            if settings.SAML2_AUTH.get('TRIGGER', {}).get('CREATE_USER', None):
-                import_string(settings.SAML2_AUTH['TRIGGER']['CREATE_USER'])(user_identity)
-            is_new_user = True
+            user_email, user_first_name, user_last_name = None, None, None
+            try:
+                user_email = user_identity[settings.SAML2_AUTH.get('ATTRIBUTES_MAP', {}).get('email', 'Email')][0]
```

Comment:
Can we just have a safer getter?

You can have a function like
```
def get_identity_attr(user_identity, key):
   if key in user_identity and user_identity[key]:
      return user_identity[key][0]
   return None
```

and use it to extract attributes

Reply:
good idea! applied in https://github.com/feedstock/django-saml2-auth/pull/4/commits/eca3ecfb9015e21a7f5cd233bc9b50ae7499687c